pylint/checkers/typecheck.py

```diff
@@ -78,6 +79,15 @@ def _unflatten(iterable):
             # We're interested only in the first element.
             yield elem
 
+def _flatten_container(iterable):
+    # Flatten nested containers into a single list
+    for item in iterable:
+        if isinstance(item, (list, tuple, types.GeneratorType)):
+            for subitem in _flatten_container(item):
```

Comment:
Since we are python 3+ only in master, this could be `yield from _flatten_container(item)`